docs/swagger_v3/oracles.spec.yaml

```diff
@@ -38,7 +38,40 @@ schemas:
       encoded_tx:
         $ref: '#/components/schemas/TransactionEncoded'
       tx:
-        type: object
+        anyOf:
+          - allOf:
+            - title: OracleRegisterTx
+            - $ref: '#/components/schemas/OracleRegisterTx'
+            - properties:
+                type:
+                  type: string
+                  example: OracleRegisterTx
+                  enum:
+                    - OracleRegisterTx
+                version:
+                  type: integer
+          - allOf:
+            - title: ContractCallTx
+            - $ref: '#/components/schemas/ContractCallTx'
+            - properties:
+                type:
+                  type: string
+                  example: ContractCallTx
+                  enum:
+                    - ContractCallTx
+                version:
+                  type: integer
+          - allOf:
+            - title: ContractCreateTx
+            - $ref: '#/components/schemas/ContractCreateTx'
+            - properties:
+                type:
+                  type: string
+                  example: ContractCreateTx
+                  enum:
+                    - ContractCreateTx
+                version:
+                  type: integer
```

Comment:
There is a simpler way in swagger docs
https://swagger.io/docs/specification/v3_0/data-models/inheritance-and-polymorphism/#mapping-type-names
This probably won't work in autorest, but would be more standard-compliant